### PR TITLE
simplify static assert

### PR DESCRIPTION
**Pull request by 0-v-0**

### CODE REVIEW

File: compiler/src/dmd/tokens.d [hunk 1]
```diff
@@ -432,11 +432,9 @@ enum FirstCKeyword = TOK.inline;
 // Assert that all token enum members have consecutive values and
 // that none of them overlap
 static assert(() {
-    foreach (idx, enumName; __traits(allMembers, TOK)) {
-       static if (idx != __traits(getMember, TOK, enumName)) {
-           pragma(msg, "Error: Expected TOK.", enumName, " to be ", idx, " but is ", __traits(getMember, TOK, enumName));
-           static assert(0);
-       }
+    static foreach (idx, enumName; __traits(allMembers, TOK)) {
```
Comment (dkorpel) [inline compiler/src/dmd/tokens.d:435]: Static foreach is not available yet in the bootstrap compiler unfortunately

Comment (dkorpel) [reply compiler/src/dmd/tokens.d:435]: Or maybe it's the multi-arguments static assert message causing the error

**Comment by 0-v-0** [comment]
What other features cannot be used in the bootstrap compiler?

**Comment by dkorpel** [comment]
Currently the bootstrap compiler is dmd 2.076, so anything introduced after that is off the table:

https://dlang.org/changelog/2.076.0.html

**Review by dkorpel — APPROVED** [review]

**Pull request merged**